qiskit/opflow/gradients/natural_gradient.py

```diff
@@ -120,33 +125,79 @@ def convert(
         def combo_fn(x):
-            c = np.real(x[0])
-            a = np.real(x[1])
-            if self.regularization:
-                # If a regularization method is chosen then use a regularized solver to
-                # construct the natural gradient.
-                nat_grad = NaturalGradient._regularized_sle_solver(
-                    a, c, regularization=self.regularization
-                )
-            else:
-                try:
-                    # Try to solve the system of linear equations Ax = C.
-                    nat_grad = np.linalg.solve(a, c)
-                except np.linalg.LinAlgError:  # singular matrix
-                    nat_grad = np.linalg.lstsq(a, c)[0]
-            return np.real(nat_grad)
+            return self.nat_grad_combo_fn(x, self.regularization)
 
         # Define the ListOp which combines the gradient and the QFI according to the combination
         # function defined above.
         return ListOp([grad, metric], combo_fn=combo_fn)
 
+    @staticmethod
+    def nat_grad_combo_fn(x: tuple, regularization: Optional[str] = None) -> np.ndarray:
+        r"""
+        Natural Gradient Function Implementation.
+
+        Args:
+            x: Iterable consisting of Gradient, Quantum Fisher Information.
+            regularization: Regularization method.
+
+        Returns:
+            Natural Gradient.
+
+        Raises:
+            ValueError: If the gradient has imaginary components that are non-negligible.
+
+        """
+        gradient = x[0]
+        metric = x[1]
+        if np.amax(np.abs(np.imag(gradient))) > ETOL:
+            raise ValueError(
+                "The imaginary part of the gradient are non-negligible. The largest absolute "
+                f"imaginary value in the gradient is {np.amax(np.abs(np.imag(gradient)))}. "
+                "Please increase the number of shots."
+            )
+        gradient = np.real(gradient)
+
+        if np.amax(np.abs(np.imag(metric))) > ETOL:
+            raise ValueError(
+                "The imaginary part of the metric are non-negligible. The largest "
+                "absolute imaginary value in the gradient is "
+                f"{np.amax(np.abs(np.imag(metric)))}. Please "
+                "increase the number of shots."
+            )
+        metric = np.real(metric)
+
+        if regularization is not None:
+            # If a regularization method is chosen then use a regularized solver to
+            # construct the natural gradient.
+            nat_grad = NaturalGradient._regularized_sle_solver(
+                metric, gradient, regularization=regularization
+            )
+        else:
+            # Check if numerical instabilities lead to a metric which is not positive semidefinite
+            w, v = np.linalg.eigh(metric)
+
+            if not all(ew >= (-1) * ETOL for ew in w):
+                raise ValueError(
+                    f"The underlying metric has ein Eigenvalue < -{ETOL}. "
```

Comment:
Should this `The underlying metric has ein Eigenvalue` be `The underlying metric has an Eigenvalue` i.e. `an` not `ein` Or perhaps even `The underlying metric has one or more Eigenvalue(s) <....`  or more simply `The underlying metric has Eigenvalue(s) <...`

Reply:
Yes, you are right. Thanks for catching that.
Apparently, I had a moment of German words in my mind :D